astropy/time/core.py

```diff
@@ -18,6 +20,12 @@
 import numpy as np
 import erfa
 
+if TYPE_CHECKING:
+    import astropy.time
+    from astropy.utils.compat import NUMPY_LT_1_20
+    if not NUMPY_LT_1_20:
+        import numpy.typing as npt
```

Comment:
I'd suggest moving this logic to ``astropy.utils.compat.numpycompat.py``

In ``astropy.utils.compat.numpycompat.py``:

```python
from types import ModuleType
from astropy.utils.compat import NUMPY_LT_1_20

if NUMPY_LT_1_20:
    npt = ModuleType("npt", doc="no-op numpy typing module")
else:
    import numpy.typing as npt
```

Here:

```suggestion
    from astropy.utils.numpycompat import npt
```